ci-operator/step-registry/ipi/conf/gcp/zones/ipi-conf-gcp-zones-commands.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+GCP_REGION="${LEASED_RESOURCE}"
+ZONES_COUNT=3
+ZONES_STR=""
+
+function join_by { local IFS="$1"; shift; echo "$*"; }
+
+function get_zones_by_machine_type() {
+  local machine_type=$1
+
+  mapfile -t AVAILABILITY_ZONES < <(gcloud compute machine-types list --filter="zone~${GCP_REGION} AND name=${machine_type}" --format='value(zone)' | sort)
+  ZONES=("${AVAILABILITY_ZONES[@]:0:${ZONES_COUNT}}")
+  ZONES_STR="[ $(join_by , "${ZONES[@]}") ]"
+  echo "[${machine_type}] GCP region: ${GCP_REGION} (zones: ${ZONES_STR})"
+}
+
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+GOOGLE_PROJECT_ID=$(jq -r .project_id ${GCP_SHARED_CREDENTIALS_FILE})
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+PATCH="${SHARED_DIR}/zones.yaml.patch"
+
+if [[ -n "${COMPUTE_NODE_TYPE}" ]]; then
```

Comment:
Can we introduce `ZONE` setting var, when user specify the var, then use the specified ones, so that make the step to be more powerful.

Reply:
`ZONES` env var, right?

Reply:
Maybe `CONTROL_PLANE_ZONES` and `COMPUTE_ZONES`

Reply:
oh, ok, so `CONTROL_PLANE_ZONES` will mutual exclusive with `CONTROL_PLANE_NODE_TYPE`, and the former takes the priority, right?

Reply:
Right.